pkg/kv/dynamodb/store.go

```diff
@@ -474,7 +474,13 @@ func (e *EntriesIterator) runQuery() {
 		e.err = fmt.Errorf("query: %w", err)
 		return
 	}
-	dynamoConsumedCapacity.WithLabelValues(operation).Add(*queryResult.ConsumedCapacity.CapacityUnits)
+
+	var ConsumedCapacity float64 = 0
+	if nil != queryResult.ConsumedCapacity {
+		ConsumedCapacity = *queryResult.ConsumedCapacity.CapacityUnits
+	}
+
+	dynamoConsumedCapacity.WithLabelValues(operation).Add(ConsumedCapacity)
```

Comment:
I would rather not publish the metric if I do not know it.
```suggestion
	if queryResult.ConsumedCapacity != nil {
		dynamoConsumedCapacity.WithLabelValues(operation).Add(*queryResult.ConsumedCapacity.CapacityUnits)
	}
```

This also brings comparison in line with the more usual Go style.

Reply:
It also align with all the other operations that we perform on dynamodb and request for consumed capacity.